wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -548,6 +549,7 @@ class KotlinGenerator private constructor(
 
     addDefaultFields(type, companionBuilder, nameAllocator)
     addAdapter(type, companionBuilder)
+    if (buildersOnly || javaInterOp) addBuildClosure(type, companionBuilder)
```

Comment:
I added support for `javaInterop` since we get Builders there too - but I don't know exactly what the implications would be if Java code tried to use this (not a Java expert). Looking for feedback there especially.

Reply:
I probably wouldn't include it for `buildersOnly` since this technically isn't a classic Java builder pattern, but would for `javaInterop` if this build closure is going to be default generated for all Kotlin targets.

It probably shouldn't be generated if the target is Java given I don't think Java supports Kotlin-style closures like this.